src/jit/lower.cpp

```diff
@@ -3841,46 +3841,266 @@ GenTree* Lowering::LowerAdd(GenTree* node)
 }
 
 //------------------------------------------------------------------------
-// LowerUnsignedDivOrMod: transform GT_UDIV/GT_UMOD nodes with a const power of 2
-// divisor into GT_RSZ/GT_AND nodes.
+// GetUnsignedMagicNumberForDivide: Generates a magic number and shift amount for
+// the magic number unsigned division optimization.
 //
 // Arguments:
-//    node - pointer to the GT_UDIV/GT_UMOD node to be lowered
+//    denom - The denominator
+//    shift - Pointer to the shift value to be returned
+//    add   - Pointer to a flag indicating the kind of code to generate
 //
-void Lowering::LowerUnsignedDivOrMod(GenTree* node)
+// Returns:
+//    The magic number.
+//
+// Notes:
+//    This code is previously from UTC where it notes it was taken from
+//   _The_PowerPC_Compiler_Writer's_Guide_, pages 57-58. The paper is is based on
+//   is "Division by invariant integers using multiplication" by Torbjorn Granlund
+//   and Peter L. Montgomery in PLDI 94
+
+template <typename U>
+U GetUnsignedMagicNumberForDivide(U denom, int* shift /*out*/, bool* add /*out*/)
```

Comment:
I'm interested to see if this shows up in throughput.   Just reading through, I was wondering if it would be worth pre-computing some set of common cases and leaving them in a table?